collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,103 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        for x, y in zip(self, other):
+            if not x == y:
+                pass
+            else:
+                return self == other
 
     def __ne__(self, other):
-        return not self == other
+        for x, y in zip(self, other):
+            if x == y:
+                pass
+            else:
+                return not self == other
 
     def __lt__(self, other):
-        pass
+        return self < other
 
     def __gt__(self, other):
-        pass
+        return self > other
 
     def __le__(self, other):
-        pass
+        return self <= other
 
     def __ge__(self, other):
-        pass
+        return self >= other
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        self.n = 0
+        self.elements = self.get_elements()
+        return(self)
 
     def __next__(self):
-        pass
+        while self.n < len(self.elements):
+            element = self.elements[self.n]
+            self.n += 1
+            return element
 
+        raise StopIteration
+            
     next = __next__
 
     def __len__(self):
-        pass
+        return len(self)
 
     def __getitem__(self, key):
-        pass
+        return self[key]
 
     def __setitem__(self, key, value):
-        pass
+        self[key] = value
 
     def __delitem__(self, key):
-        pass
+        del self[key]
 
     def __contains__(self, item):
-        pass
+        return item in self.elements
 
 
 class RepresentableMixin(object):
     def __repr__(self):
-        pass
+        return str(self.elements)
 
     def __str__(self):
-        pass
+        return str(self.elements)
 
 
 class ConstructibleMixin(object):
     DATA_ATTR_NAME = 'data'
 
```

Comment:
It looks like you made things easier by defining self.elements in terms of the get elements function.  But I am curious where self.data comes from.